src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/HarvestCropsBehavior.java

```diff
@@ -93,15 +91,25 @@ private static boolean isBlockCrops(ItemStack stack, Level world, Player player,
     }
 
     private static boolean harvestBlockRoutine(ItemStack stack, BlockPos pos, Player player) {
-        BlockState blockState = player.level().getBlockState(pos);
-        Block block = blockState.getBlock();
-        CropBlock blockCrops = (CropBlock) block;
-        if (blockCrops.isMaxAge(blockState)) {
-            NonNullList<ItemStack> drops = NonNullList.create();
-            drops.addAll(Block.getDrops(blockState, (ServerLevel) player.level(), pos, null));
-            dropListOfItems(player.level(), pos, drops);
-            player.level().levelEvent(LevelEvent.PARTICLES_DESTROY_BLOCK, pos, Block.getId(blockState));
-            player.level().setBlock(pos, blockCrops.getStateForAge(0), Block.UPDATE_ALL);
+        var level = player.level();
+        var blockState = level.getBlockState(pos);
+        var block = blockState.getBlock();
+        var cropBlock = (CropBlock) block;
+        final var seed = cropBlock.getCloneItemStack(level, pos, blockState).getItem();
+        if (cropBlock.isMaxAge(blockState)) {
+            final var check = new boolean[] { true };
+            var drops = Block.getDrops(blockState, (ServerLevel) level, pos, null).stream()
+                    .peek(itemStack -> {
+                        if (check[0] && itemStack.is(seed)) {
+                            itemStack.shrink(1);
+                            check[0] = false;
+                        }
+                    })
+                    .filter(itemStack -> itemStack.getCount() > 0)
+                    .toList();
```

Comment:
As an avid user of streams: this is cursed. Just use a `listIterator` to search for the seed stack, then remove it if empty after the shrink. Then break.